### PR TITLE
Fix views on Vanilla PG and add tests

### PR DESCRIPTION
**Pull request by cevian**

### CODE REVIEW

File: pkg/pgmodel/testdata/expected/views.out [hunk 1]
```diff
@@ -1,22 +1,20 @@
 \set ON_ERROR_STOP 1
 SELECT _prom_catalog.get_or_create_metric_table_name('cpu_usage');
-psql:/testdata/sql/views.sql:4: NOTICE:  adding not-null constraint to column "time"
-DETAIL:  Time dimensions cannot have NULL values
-psql:/testdata/sql/views.sql:4: NOTICE:  adding index _compressed_hypertable_2_series_id__ts_meta_sequence_num_idx ON _timescaledb_internal._compressed_hypertable_2 USING BTREE(series_id, _ts_meta_sequence_num)
  get_or_create_metric_table_name 
 ---------------------------------
- (1,cpu_usage)
+ (1,cpu_usage,t)
```
Comment (JLockerman) [inline pkg/pgmodel/testdata/expected/views.out:5]: how come this changed? I don't see anything in the SQL that should have changed it

**Comment by JLockerman** [comment]
In general looks good to me, will rereview in the morning when awake

File: pkg/pgmodel/testdata/expected/views.out (hunk 1 above)
Comment (cevian) [reply pkg/pgmodel/testdata/expected/views.out:5]: This test was silently broken for a while (see first commit) so the golden file was out of date.

**Review by antekresic — APPROVED** [review]
Looks good :+1:

File: pkg/pgmodel/testdata/expected/views.out (hunk 1 above)
Comment (JLockerman) [reply pkg/pgmodel/testdata/expected/views.out:5]: ah, ouch.

**Review by JLockerman — APPROVED** [review]

**Pull request merged**